module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -2147,3 +2170,28 @@ public function getNewItems($page, $limit, $daysOld, $fundId = null)
         return [];
     }
 }
+
+/**
+ * Invisible Availability Status
+ *
+ * Items with this availability status will not be visible in the list of holdings.
+ *
+ * @category VuFind
+ * @package  ILS_Drivers
+ * @author   Peter Murray <peter@indexdata.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:ils_drivers?s[]=availabilitystatus Wiki
+ */
+class InvisibleAvailabilityStatus extends AvailabilityStatus
```

Comment:
You should move this to its own file in the VuFind\ILS\Logic namespace; this isn't FOLIO-specific and might be useful for other drivers, so moving it would make it more reusable. We also have a "one class per file" rule in the style checker, so this move should help get the build passing as well.

Reply:
Alternatively, though, I wonder if it would make sense to add a new setter and/or constructor parameter to the existing AvailabilityStatus class to allow visibility toggling/configuration instead of requiring a new subclass at all. I wonder if @ThoWagen has any opinions about this.

Reply:
Yes, I think adding a setter and getter in the base class makes sense.

Reply:
I did it this way because the intention of the class as written seemed to be to [override the `isVisibleInHoldings()` method](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/ILS/Logic/AvailabilityStatus.php#L89-L98), but I can reconfigure the class if necessary. (In the meantime, I pulled this class out of the Driver and into Logic.)

Reply:
Since @ThoWagen wrote the original class but agrees with the idea of adding a getter and setter, that seems to be justification to make that change instead of adding a new class. :-)

Reply:
Good enough for me! I'm adding this and removing the new class:
```php
    /**
     * Set visibility in holdings tab.
     */
    public function setVisibility(bool $visibility): AvailabilityStatus
    {
        $this->visibility = $visibility;
        return $this;
    }
```